sql/src/main/java/io/crate/execution/engine/profile/CollectProfileOperation.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.execution.engine.profile;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+public interface CollectProfileOperation {
+
+    /**
+     * @param nodeId       the nodeId of the node from which to collect profiling information from
+     * @param closeContext indicate if context must be closed after fetch
```

Comment:
I guess this was copied from the fetch stuff. This flag shouldn't be required, we can close the context after the first request. In the fetch-case we support streaming results which is why there may be multiple fetch calls. But this here should only be done once after the actual operation finished.

Reply:
Yes it was. Ok, got it.